tests/unit/test-cmd/test.build.js

```diff
@@ -48,6 +53,50 @@ describe('build', () => {
     );
   });
 
+  it('ignore artifacts dir if included by source dir when zipping', () => {
```

Comment:
rejoice! Since you moved all of the ignoring logic, you no longer need such a heavyweight test. Here's what you can do:
* you can delete all of the fixture directories (with the manifest, etc)
* you can simply assert that build is configuring the file watcher correctly with its value of `ignoreFiles`. Here are [some examples](https://github.com/mozilla/web-ext/blob/master/tests/unit/test-cmd/test.build.js#L249) of how we do things like that.